executor/oci/mounts.go

```diff
@@ -49,20 +52,69 @@ func GetMounts(ctx context.Context, mountOpts ...MountOpts) []specs.Mount {
 			Options:     []string{"nosuid", "noexec", "nodev", "ro"},
 		},
 	}
+	var err error
 	for _, o := range mountOpts {
-		mounts = o(mounts)
+		mounts, err = o(mounts)
+		if err != nil {
+			return nil, err
+		}
 	}
-	return mounts
+	return mounts, nil
 }
 
-func withROBind(src, dest string) func(m []specs.Mount) []specs.Mount {
-	return func(m []specs.Mount) []specs.Mount {
+func withROBind(src, dest string) func(m []specs.Mount) ([]specs.Mount, error) {
+	return func(m []specs.Mount) ([]specs.Mount, error) {
 		m = append(m, specs.Mount{
 			Destination: dest,
 			Type:        "bind",
 			Source:      src,
 			Options:     []string{"rbind", "ro"},
 		})
-		return m
+		return m, nil
+	}
+}
+
+func under(prefixDir, p string) bool {
+	prefixDir = filepath.Clean(prefixDir)
+	p = filepath.Clean(p)
+	return p == prefixDir || strings.HasPrefix(p, prefixDir+"/")
+}
+
+func withoutUnder(prefixDir string) func([]specs.Mount) ([]specs.Mount, error) {
```

Comment:
nit: this is a bit weird construction. `removeMountsWithPrefix([]mounts, pfx) []mounts` ?